### ISSUE TITLE
Bad request (400) - could not determine if we should add owner ref

### ISSUE THREAD

**Issue by jmazzitelli** [operator-framework/operator-sdk#1330]
## Bug Report

Let me state up front that I'm not 100% sure this is a bug - but I am reporting this as one just in case. I am willing to be told I'm doing something wrong though :-) However, if I run my playbook outside of the operator framework (i.e. I just run the ansible playbook locally), it all runs fine. So my ansible tasks and everything are OK and working as I would expect. It is only when I deploy the ansible playbook within the operator and have the operator kick off my playbook with the creation of my CR that I see the problem

**What did you do?**

Moved up to operator sdk base image of 0.7.0 (quay.io/operator-framework/ansible-operator:v0.7.0)

I see this error when my Ansible operator tries to create some custom resources (the CRD was created by the operator earlier - so this should have succeeded), however, all the custom resources fail with this error msg from this code that was added recently to ansible/proxy/proxy.go:

https://github.com/operator-framework/operator-sdk/pull/1148/files#diff-8c0ad10c79b2eb7738d725c05b5c927cR268

The logged message in ansible container is: 

```
failed: [localhost] (item=/opt/ansible/roles/kiali-deploy/templates/dashboards/vertx-server.yaml) => {"changed": false, "error": 400, "item": "/opt/ansible/roles/kiali-deploy/templates/dashboards/vertx-server.yaml", "msg": "Failed to create object: Could not determine if we should add owner ref\n", "reason": "Bad Request", "status": 400}
```

The logged message in the operator container is:

```

{"level":"info","ts":1555431572.0016534,"logger":"proxy","msg":"Injecting owner reference"}
--
  | {"level":"error","ts":1555431572.001834,"logger":"proxy","msg":"Could not get rest mapping for: monitoring.kiali.io/v1alpha1, Kind=MonitoringDashboard","error":"no matches for kind \"MonitoringDashboard\" in version \"monitoring.kiali.io/v1alpha1\"","stacktrace":"github.com/operator-framework/operator-sdk/vendor/github.com/go-logr/zapr.(*zapLogger).Error\n\tsrc/github.com/operator-framework/operator-sdk/vendor/github.com/go-logr/zapr/zapr.go:128\ngithub.com/operator-framework/operator-sdk/pkg/ansible/proxy.shouldAddOwnerRef\n\tsrc/github.com/operator-framework/operator-sdk/pkg/ansible/proxy/proxy.go:336\ngithub.com/operator-framework/operator-sdk/pkg/ansible/proxy.InjectOwnerReferenceHandler.func1\n\tsrc/github.com/operator-framework/operator-sdk/pkg/ansible/proxy/proxy.go:266\nnet/http.HandlerFunc.ServeHTTP\n\t/home/travis/.gimme/versions/go1.10.3.linux.amd64/src/net/http/server.go:1947\ngithub.com/operator-framework/operator-sdk/pkg/ansible/proxy.CacheResponseHandler.func1\n\tsrc/github.com/operator-framework/operator-sdk/pkg/ansible/proxy/proxy.go:184\nnet/http.HandlerFunc.ServeHTTP\n\t/home/travis/.gimme/versions/go1.10.3.linux.amd64/src/net/http/server.go:1947\nnet/http.serverHandler.ServeHTTP\n\t/home/travis/.gimme/versions/go1.10.3.linux.amd64/src/net/http/server.go:2694\nnet/http.(*conn).serve\n\t/home/travis/.gimme/versions/go1.10.3.linux.amd64/src/net/http/server.go:1830"}
  | {"level":"error","ts":1555431572.0018978,"logger":"proxy","msg":"Could not determine if we should add owner ref","error":"no matches for kind \"MonitoringDashboard\" in version \"monitoring.kiali.io/v1alpha1\"","stacktrace":"github.com/operator-framework/operator-sdk/vendor/github.com/go-logr/zapr.(*zapLogger).Error\n\tsrc/github.com/operator-framework/operator-sdk/vendor/github.com/go-logr/zapr/zapr.go:128\ngithub.com/operator-framework/operator-sdk/pkg/ansible/proxy.InjectOwnerReferenceHandler.func1\n\tsrc/github.com/operator-framework/operator-sdk/pkg/ansible/proxy/proxy.go:269\nnet/http.HandlerFunc.ServeHTTP\n\t/home/travis/.gimme/versions/go1.10.3.linux.amd64/src/net/http/server.go:1947\ngithub.com/operator-framework/operator-sdk/pkg/ansible/proxy.CacheResponseHandler.func1\n\tsrc/github.com/operator-framework/operator-sdk/pkg/ansible/proxy/proxy.go:184\nnet/http.HandlerFunc.ServeHTTP\n\t/home/travis/.gimme/versions/go1.10.3.linux.amd64/src/net/http/server.go:1947\nnet/http.serverHandler.ServeHTTP\n\t/home/travis/.gimme/versions/go1.10.3.linux.amd64/src/net/http/server.go:2694\nnet/http.(*conn).serve\n\t/home/travis/.gimme/versions/go1.10.3.linux.amd64/src/net/http/server.go:1830"}
```

NOTE: the operator is installed in namespace 'kiali-operator' and the custom resources (and their CRD) is created in another namespace ('istio-system'). I suspect this might have something to do with it, based on the comments I see in the PR #1148 that introduced that change to ansible/proxy/proxy.go

**What did you expect to see?**

The custom resources created successfully.

**Environment**
* operator-sdk version: 0.7.0

The Dockerfile to build my ansible operator uses the 0.7.0 base image.

* Kubernetes version information:

OpenShift 3.11

* Are you writing your operator in ansible, helm, or go?

ansible

**Comment by jmazzitelli** [comment]
@shawn-hurley the more I play with this, the more I think this is a bug. But you tell me what you think after you look at it.

**Comment by shawn-hurley** [comment]
@jmazzitelli I notice this line:  

```{"level":"error","ts":1555431572.001834,"logger":"proxy","msg":"Could not get rest mapping for: monitoring.kiali.io/v1alpha1, Kind=MonitoringDashboard","error":"no matches for kind \"MonitoringDashboard\" in version \"monitoring.kiali.io/v1alpha1\""
```

Which makes me believe that the operator does not know about that type. Can you please give me more background on how that task and the preceding tasks work?

**Comment by jmazzitelli** [comment]
> @jmazzitelli I notice this line:
> 
> Which makes me believe that the operator does not know about that type. Can you please give me more background on how that task and the preceding tasks work?

@shawn-hurley That is a blank line - I assume you meant to enter something here? :)

**Comment by jmazzitelli** [comment]
@shawn-hurley regarding this error:

```
{"level":"error","ts":1555431572.001834,"logger":"proxy","msg":"Could not get rest mapping for: 
monitoring.kiali.io/v1alpha1, Kind=MonitoringDashboard","error":"no matches for kind 
\"MonitoringDashboard\" in version \"monitoring.kiali.io/v1alpha1\""
```

This is the CRD the ansible operator creates in the target namespace (NOT the namespace where the operator is - in the target namespace where the operator is putting Kiali):

https://github.com/kiali/kiali/blob/master/operator/roles/kiali-deploy/templates/openshift/crds.yaml

```
spec:
  group: monitoring.kiali.io
  names:
    kind: MonitoringDashboard
...
  version: v1alpha1
```
(this is created earlier in the Ansible role in this task [found in os-main.yml](https://github.com/kiali/kiali/blob/master/operator/roles/kiali-deploy/tasks/openshift/os-main.yml#L23)  which was called from [main.yml](https://github.com/kiali/kiali/blob/master/operator/roles/kiali-deploy/tasks/main.yml#L122-L125)

These are the templates for all the CRs it creates:

https://github.com/kiali/kiali/blob/master/operator/roles/kiali-deploy/templates/dashboards

They all use: `apiVersion: "monitoring.kiali.io/v1alpha1"` as you see [here](https://github.com/kiali/kiali/blob/master/operator/roles/kiali-deploy/templates/dashboards/go.yaml#L1) for example

These are created [in main.yml](https://github.com/kiali/kiali/blob/master/operator/roles/kiali-deploy/tasks/main.yml#L132-L139), after the CRD is created.

One thing I thought might be the problem is the CRD isn't ready yet by the time the CRs are created, but I have a PR I've been working with that waits for the CRD to be "Established" (see [here](https://github.com/kiali/kiali/pull/1010/files#diff-0f908776f2b67d4a2fd285d5a852468d)), but I still see the error.

**Comment by shawn-hurley** [comment]
Ok, so this is a problem that is known and there is a fix here: https://github.com/operator-framework/operator-sdk/pull/1329

Basically, the issue is that the restmapper is created at startup of the operator. The rest mapper finds all the resources in the cluster so that we can make smart decisions about how to interact with the cluster. Because it appears that you are creating the CRD in your reconciliation loop, this will cause the restmapper to never know about the CRD. I bet that after the CRD is created and you restart the operator that you would see the task complete. 

Is there a reason that you can not create the CRD before hand? I think it is usually unwise to give an operator permissions to create the CRD and I usually advise against this pattern. 

Once the PR merges above we can update the ansible operator to use that rest mapper and it should fix this issue. I will leave it open to track this.

**Comment by jmazzitelli** [comment]
> Is there a reason that you can not create the CRD before hand? I think it is usually unwise to give an operator permissions to create the CRD and I usually advise against this pattern.

Well, this CRD is very specific to Kiali. It is a way to have Kiali be able to load in custom dashboards (users can later add their own if they want, but Kiali comes with some out of the box).

The user is supposed to use the operator to install everything related to a product, no? This CRD is part of that product (Kiali). It would seem kind of odd to ask the user to install a CRD ahead of time in order for the operator to properly install the product. Unless I am not understanding the overall purpose of operators and how they are to be used. I'm trying to reduce the amount of steps people need to do to install Kiali - seems counter-productive to say, "in order to install Kiali, you have to (1) install the Kiali dashbard CRD, and then (2) install the Operator, and then (3) you can install the Kiali CR to kick off the install.

In other words, if a product itself requires CRDs, how are users supposed to do this with operators?

**Comment by jcrossley3** [comment]
I think as more and more people start to build k8s-native apps (written in terms of CRDs + controllers) it will be more common to expect operators to install them.

**Comment by jmazzitelli** [comment]
> Ok, so this is a problem that is known and there is a fix here: #1329

I just used 0.8.0 (which should have that fix) and I am still seeing the problem:

Make sure I'm running 0.8.0:

```
{"level":"info","ts":1558465126.8952413,"logger":"cmd","msg":"Go Version: go1.12.5"}
{"level":"info","ts":1558465126.8952928,"logger":"cmd","msg":"Go OS/Arch: linux/amd64"}
{"level":"info","ts":1558465126.895298,"logger":"cmd","msg":"Version of operator-sdk: v0.8.0"}
{"level":"info","ts":1558465126.895325,"logger":"cmd","msg":"Watching namespace.","Namespace":"kiali-operator"}
```

But I still see the error:

```
{"level":"error","ts":1558465170.3098834,"logger":"proxy","msg":"Could not get rest mapping for: monitoring.kiali.io/v1alpha1, Kind=MonitoringDashboard","error":"no matches for kind \"MonitoringDashboard\" in version \"monitoring.kiali.io/v1alpha1\"","stacktrace":"github.com/operator-framework/operator-sdk/vendor/github.com/go-logr/zapr.(*zapLogger).Error\n\tsrc/github.com/operator-framework/operator-sdk/vendor/github.com/go-logr/zapr/zapr.go:128\ngithub.com/operator-framework/operator-sdk/pkg/ansible/proxy.shouldAddOwnerRef\n\tsrc/github.com/operator-framework/operator-sdk/pkg/ansible/proxy/inject_owner.go:151\ngithub.com/operator-framework/operator-sdk/pkg/ansible/proxy.(*injectOwnerReferenceHandler).ServeHTTP\n\tsrc/github.com/operator-framework/operator-sdk/pkg/ansible/proxy/inject_owner.go:89\ngithub.com/operator-framework/operator-sdk/pkg/ansible/proxy.(*cacheResponseHandler).ServeHTTP\n\tsrc/github.com/operator-framework/operator-sdk/pkg/ansible/proxy/cache_response.go:127\nnet/http.serverHandler.ServeHTTP\n\t/home/travis/.gimme/versions/go1.12.5.linux.amd64/src/net/http/server.go:2774\nnet/http.(*conn).serve\n\t/home/travis/.gimme/versions/go1.12.5.linux.amd64/src/net/http/server.go:1878"}
{"level":"error","ts":1558465170.309941,"logger":"proxy","msg":"Could not determine if we should add owner ref","error":"no matches for kind \"MonitoringDashboard\" in version \"monitoring.kiali.io/v1alpha1\"","stacktrace":"github.com/operator-framework/operator-sdk/vendor/github.com/go-logr/zapr.(*zapLogger).Error\n\tsrc/github.com/operator-framework/operator-sdk/vendor/github.com/go-logr/zapr/zapr.go:128\ngithub.com/operator-framework/operator-sdk/pkg/ansible/proxy.(*injectOwnerReferenceHandler).ServeHTTP\n\tsrc/github.com/operator-framework/operator-sdk/pkg/ansible/proxy/inject_owner.go:92\ngithub.com/operator-framework/operator-sdk/pkg/ansible/proxy.(*cacheResponseHandler).ServeHTTP\n\tsrc/github.com/operator-framework/operator-sdk/pkg/ansible/proxy/cache_response.go:127\nnet/http.serverHandler.ServeHTTP\n\t/home/travis/.gimme/versions/go1.12.5.linux.amd64/src/net/http/server.go:2774\nnet/http.(*conn).serve\n\t/home/travis/.gimme/versions/go1.12.5.linux.amd64/src/net/http/server.go:1878"}
```

**Comment by jmazzitelli** [comment]
In addition, we (in Kiali) are also seeing this "cannot determine owner" error when attempting to create an OpenShift route (not related to CRDs). I'm not sure if its related or not, but see: https://issues.jboss.org/browse/KIALI-2873

**Comment by jmazzitelli** [comment]
> Once the PR merges above we can update the ansible operator to use that rest mapper and it should fix this issue. I will leave it open to track this.

I didn't notice this final sentence. So now I understand we need to wait for the ansible operator to use this new stuff.

**Comment by slopezz** [comment]
Although PR have been already merged, I still have the same error.

I have a sample ansible hello-world operator created using source operator-sdk version 0.6.0, which is working correctly (it creates the following objects: ImageStream, DeploymentCondig, Service, Route).

If I create the same ansible operator with new source operator-sdk version 0.8.0 or 0.8.1, I get the following error when trying to create the first object:
```
TASK [hello-world : Manage hello-world ImageStream for CR example-helloworld on project example] ***
task path: /opt/ansible/roles/hello-world/tasks/main.yml:11
fatal: [localhost]: FAILED! => {"changed": false, "error": 400, "msg": "Failed to create object: Could not determine if we should add owner ref\n", "reason": "Bad Request", "status": 400}
```

However:
- If I use my sample ansible hello-world operator with source operator-sdk v0.6.0, objects are created correctly (OK)
- Then, with already created objects, I update sample hello-world operator with source operator-sdk v0.8.0 or v0.8.1
- Now reconcile works correctly once objects are already created (all tasks finish with OK status), so the problem appears when creating objects

**Comment by jmazzitelli** [comment]
@slopezz read this: https://github.com/operator-framework/operator-sdk/pull/1473

The original PR that was merged wasn't the only thing needed - the ansible operator framework needs that PR #1473 , too (to address my specific issue reported in this issue - not sure if it is your issue, too). I do not believe that second PR is going to be merged.

The short answer is - you need to not create CRDs in the operator.

The operator-sdk team is **strongly** recommending that operators should not create CRDs. Rather than swim upstream, I am no longer doing that in my operator - my operator no longer creates the CRDs it needs and I am documenting that my users need to create the additional CRDs my app needs (in addition to the operator CRD itself). In other words, operators require all CRDs to be created already. At that point, the operator can create CRs without issue.

**Comment by slopezz** [comment]
@jmazzitelli I'm not creating the CRD with the operator, what I am doing is the following (in 3 different steps):

* First I create CRD:
    - crd.yaml

* Secondly I create the operator:
   - service_account.yaml
   - role.yaml
   - role_binding.yaml
   - operator.yaml

* Finally, I create a specific CR (or N CRs):
   - cr.yaml

* So now operator read CR (or CRs), and created desired objects.

So when using my ansible operator with source `operator-sdk v0.8.0/v0.8.1`, when trying to create the first object I got that error `Failed to create object: Could not determine if we should add owner ref\n`

But:
- If I first create everything with my operator using `operator-sdk v0.6.0`, which creates correctly all desired objects based on CR spec.
- Then I upgrade that operator deployment to use new `operator-sdk v0.8.0/v0.8.1` (by setting the new image on the current deployed operator.yaml Deployment).
- Now I can see that reconcile works as expected (because objects have previously been created when operator was using `previous v0.6.0`), and all tasks finish with `ok` status (I guess it works because it can read current object `ownerReferences metadata` field).
- But If I delete manually any created object, on the next reconcile period operator won't be able to recreate the missing object, because getting error `Failed to create object: Could not determine if we should add owner ref\n`.

**Comment by shawn-hurley** [comment]
In v0.6.0 we were doing some incorrect things with owner references. Now that we are doing the correct things, it sounds like something is unable to find some info about the type that you are attempting to create in your operator. 

There should be another error log saying "could not get (group version/rest mapping) for xxxx" can you tell me what that is saying?

**Comment by slopezz** [comment]
Yes @shawn-hurley , on operator container log there is the following message:

```
{"level":"error","ts":1560415815.2956142,"logger":"proxy","msg":"Could not get rest mapping for: /v1, Kind=ImageStream","error":"no matches for kind \"ImageStream\" in version \"v1\"","stacktrace":"github.com/operator-framework/operator-sdk/vendor/github.com/go-logr/zapr.(*zapLogger).Error\n\tsrc/github.com/operator-framework/operator-sdk/vendor/github.com/go-logr/zapr/zapr.go:128\ngithub.com/operator-framework/operator-sdk/pkg/ansible/proxy.shouldAddOwnerRef\n\tsrc/github.com/operator-framework/operator-sdk/pkg/ansible/proxy/inject_owner.go:151\ngithub.com/operator-framework/operator-sdk/pkg/ansible/proxy.(*injectOwnerReferenceHandler).ServeHTTP\n\tsrc/github.com/operator-framework/operator-sdk/pkg/ansible/proxy/inject_owner.go:89\ngithub.com/operator-framework/operator-sdk/pkg/ansible/proxy.(*cacheResponseHandler).ServeHTTP\n\tsrc/github.com/operator-framework/operator-sdk/pkg/ansible/proxy/cache_response.go:127\nnet/http.serverHandler.ServeHTTP\n\t/home/travis/.gimme/versions/go1.12.5.linux.amd64/src/net/http/server.go:2774\nnet/http.(*conn).serve\n\t/home/travis/.gimme/versions/go1.12.5.linux.amd64/src/net/http/server.go:1878"}

{"level":"error","ts":1560415815.2957528,"logger":"proxy","msg":"Could not determine if we should add owner ref","error":"no matches for kind \"ImageStream\" in version \"v1\"","stacktrace":"github.com/operator-framework/operator-sdk/vendor/github.com/go-logr/zapr.(*zapLogger).Error\n\tsrc/github.com/operator-framework/operator-sdk/vendor/github.com/go-logr/zapr/zapr.go:128\ngithub.com/operator-framework/operator-sdk/pkg/ansible/proxy.(*injectOwnerReferenceHandler).ServeHTTP\n\tsrc/github.com/operator-framework/operator-sdk/pkg/ansible/proxy/inject_owner.go:92\ngithub.com/operator-framework/operator-sdk/pkg/ansible/proxy.(*cacheResponseHandler).ServeHTTP\n\tsrc/github.com/operator-framework/operator-sdk/pkg/ansible/proxy/cache_response.go:127\nnet/http.serverHandler.ServeHTTP\n\t/home/travis/.gimme/versions/go1.12.5.linux.amd64/src/net/http/server.go:2774\nnet/http.(*conn).serve\n\t/home/travis/.gimme/versions/go1.12.5.linux.amd64/src/net/http/server.go:1878"}

{"level":"error","ts":1560415815.3653424,"logger":"logging_event_handler","msg":"","name":"example-helloworld","namespace":"example","gvk":"hello-world-operator.com/v1alpha1, Kind=HelloWorld","event_type":"runner_on_failed","job":"3588944213631772581","EventData.Task":"Manage hello-world ImageStream for CR example-helloworld on project example","EventData.TaskArgs":"_ansible_version=2.7.10, definition=apiVersion: v1\nkind: ImageStream\nmetadata:\n  name: \"example-helloworld-hello-world\"\n  namespace: \"example\"\nspec:\n  tags:\n    - name: \"2.0\"\n      annotations:\n        openshift.io/display-name: hello-world 2.0\n      from:\n        kind: DockerImage\n        name: \"gcr.io/google-samples/hello-app:2.0\"\n      referencePolicy:\n        type: Source\n      importPolicy: {}\n    - name: latest\n      annotations:\n        openshift.io/display-name: hello-world latest\n      from:\n        kind: ImageStreamTag\n        name: \"2.0\"\n      referencePolicy:\n        type: Source\n      importPolicy: {}\n, _ansible_selinux_special_fs=['fuse', 'nfs', 'vboxsf', 'ramfs', '9p'], _ansible_no_log=False, _ansible_module_name=k8s, _ansible_debug=False, _ansible_verbosity=2, _ansible_keep_remote_files=False, _ansible_syslog_facility=LOG_USER, _ansible_socket=None, _ansible_diff=False, _ansible_remote_tmp=~/.ansible/tmp, _ansible_shell_executable=/bin/sh, _ansible_check_mode=False, _ansible_tmpdir=/opt/ansible/.ansible/tmp/ansible-tmp-1560415813.88-252922631565951/","EventData.FailedTaskPath":"/opt/ansible/roles/hello-world/tasks/main.yml:3","error":"[playbook task failed]","stacktrace":"github.com/operator-framework/operator-sdk/vendor/github.com/go-logr/zapr.(*zapLogger).Error\n\tsrc/github.com/operator-framework/operator-sdk/vendor/github.com/go-logr/zapr/zapr.go:128\ngithub.com/operator-framework/operator-sdk/pkg/ansible/events.loggingEventHandler.Handle\n\tsrc/github.com/operator-framework/operator-sdk/pkg/ansible/events/log_events.go:84"}

{"level":"error","ts":1560415815.5271914,"logger":"runner","msg":"ansible-playbook 2.7.10\r\n config file = /etc/ansible/ansible.cfg\r\n configured module search path = [u'/usr/share/ansible/openshift']\r\n ansible python module location = /usr/lib/python2.7/site-packages/ansible\r\n executable location = /usr/bin/ansible-playbook\r\n python version = 2.7.5 (default, Oct 30 2018, 23:45:53) [GCC 4.8.5 20150623 (Red Hat 4.8.5-36)]\r\nUsing /etc/ansible/ansible.cfg as config file\r\n\n/tmp/ansible-operator/runner/hello-world-operator.com/v1alpha1/HelloWorld/example/example-helloworld/inventory/hosts did not meet host_list requirements, check plugin documentation if this is unexpected\r\n/tmp/ansible-operator/runner/hello-world-operator.com/v1alpha1/HelloWorld/example/example-helloworld/inventory/hosts did not meet script requirements, check plugin documentation if this is unexpected\r\n\n/tmp/ansible-operator/runner/hello-world-operator.com/v1alpha1/HelloWorld/example/example-helloworld/inventory/hosts did not meet script requirements, check plugin documentation if this is unexpected\n\r\nPLAYBOOK: 3235e58ae17843d08e44068bfcde6db2 *************************************\n1 plays in /tmp/ansible-operator/runner/hello-world-operator.com/v1alpha1/HelloWorld/example/example-helloworld/project/3235e58ae17843d08e44068bfcde6db2\n\r\nPLAY [localhost] ***************************************************************\n\r\nTASK [Gathering Facts] *********************************************************\nok: [localhost]\nMETA: ran handlers\n\r\nTASK [hello-world : Manage hello-world ImageStream for CR example-helloworld on project example] ***\r\ntask path: /opt/ansible/roles/hello-world/tasks/main.yml:3\nfatal: [localhost]: FAILED! => {\"changed\": false, \"error\": 400, \"msg\": \"Failed to create object: Could not determine if we should add owner ref\\n\", \"reason\": \"Bad Request\", \"status\": 400}\n\r\nPLAY RECAP *********************************************************************\r\nlocalhost : ok=1 changed=0 unreachable=0 failed=1 \r\n\n","job":"3588944213631772581","name":"example-helloworld","namespace":"example","error":"exit status 2","stacktrace":"github.com/operator-framework/operator-sdk/vendor/github.com/go-logr/zapr.(*zapLogger).Error\n\tsrc/github.com/operator-framework/operator-sdk/vendor/github.com/go-logr/zapr/zapr.go:128\ngithub.com/operator-framework/operator-sdk/pkg/ansible/runner.(*runner).Run.func1\n\tsrc/github.com/operator-framework/operator-sdk/pkg/ansible/runner/runner.go:294"}
```

It complains exactly about ImageStream Openshift object (created with the first ansible task):
```
Could not get rest mapping for: /v1, Kind=ImageStream","error":"no matches for kind \"ImageStream\" in version \"v1\"
```

I don't know if maybe I am setting incorrect role permission regarding this specific Object:
```
- apiGroups:
  - image.openshift.io
  resources:
  - imagestreams
  - imagestreams/layers
  verbs:
  - '*'
```
But It is weird because for testing purpose I set at the end of `role.yaml` full `*` permission:
```
- apiGroups:
  - '*'
  resources:
  - '*'
  verbs:
  - '*'
```

Here https://github.com/slopezz/ansible-hello-world-operator/tree/upgrade-operator-sdk-v0.8.1  is the code I am running (branch `upgrade-operator-sdk-v0.8.1`) where you can check all involved yaml files I'm using, there is a Makefile with target `make all` that creates example namespace, CRD, all operator objects (service_account, role, role_bindinf, operator) and CR.

I'm running it on a `OCP cluster v3.11`

**Comment by shawn-hurley** [comment]
If you notice the group that the operator is attempting to get is "" while the actual group is "image.openshift.io". If you update the k8s module definition to `apiVersion: image.openshift.io/v1` then this should fix this issue.

**Comment by slopezz** [comment]
Thank you @shawn-hurley , after changing the apiVersion to specific Openshift ones, it is working correctly again.

I saw that specifying Openshift APIs was needed on helm-operator, but as it was working well on ansible-operator v0.6.0 without specifying them, I thought it was the correct way to do so.

To know which Openshift apiGroups/resources specify on operator role.yaml, I've seen that for example [here](https://docs.openshift.com/container-platform/3.11/rest_api/apis-build.openshift.io/v1.Build.html) there are all available APIs, but what about the specific resources? I mean, for example, I've seen on different role.yaml examples, that in order to enable `Openshift Route` object permission, it is needed to specify apiGroup `route.openshift.io` (OK), but also specific resources `routes, routes/custom-host, routes/status`, and the same for example with `ImageStream` object permission, apiGroup `image.openshift.io`, and resources `imagestreams, imagestreams/layers`:
```
- apiGroups:
  - route.openshift.io
  resources:
  - routes
  - routes/custom-host
  - routes/status
  verbs:
- '*'
- apiGroups:
  - image.openshift.io
  resources:
  - imagestreams
  - imagestreams/layers
  verbs:
- '*'
```

 Is there any documentation where It can be checked which apiGroup and mainly, which resources are needed to be  specified on operator role  for every specific Openshit object?
